File: leetcode/editor/en/[13]Roman to Integer.java
13
        Roman to Integer
        2023-01-01 01:03:17

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int romanToInt(String s) {
        int res = 0;
        int num = 0;
        int temp = 0;
        for (int i = s.length() - 1; i >= 0; i--) {
            switch (s.charAt(i)) {
                case 'I' -> num = 1;
                case 'V' -> num = 5;
                case 'X' -> num = 10;
                case 'L' -> num = 50;
                case 'C' -> num = 100;
                case 'D' -> num = 500;
                case 'M' -> num = 1000;
            }
            if (num < temp) {
                res -= num;
            } else {
                res += num;
            }
            temp = num;
        }
        return res;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[13]Roman to Integer.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function (s) {
    let res = 0;
    let num = 0;
    let temp = 0;
    for (let i = s.length - 1; i >= 0; i--) {
        switch (s.charAt(i)) {
            case 'I':
                num = 1;
                break;
            case 'V':
                num = 5;
                break;
            case 'X':
                num = 10;
                break;
            case 'L':
                num = 50;
                break;
            case 'C':
                num = 100;
                break;
            case 'D':
                num = 500;
                break;
            case 'M':
                num = 1000;
                break;
        }
        if (num < temp) {
            res -= num;
        } else {
            res += num;
        }
        temp = num;
    }
    return res;
};
//leetcode submit region end(Prohibit modification and deletion)
